File: solution/python/242.py
# Use two dictionary to compare the frequency of each character.
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        sDict = {}
        tDict = {}
        for i in s:
            if i not in sDict:
                sDict[i] = 1
            else:
                sDict[i] += 1
        for i in t:
            if i not in tDict:
                tDict[i] = 1
            else:
                tDict[i] += 1
        if sDict == tDict:
            return True
        else:
            return False

File: solution/go/242.go
/*
Use hashmap to store the frequency of each char
*/
func isAnagram(s string, t string) bool {
    stats := make(map[rune]int)
    if len(s) > len(t) {
        t, s = s, t
    }
    for _, value := range s {
        stats[value] += 1
    }
    for _, value := range t {
        if val, ok := stats[value]; ok && val > 0 {
            stats[value] -= 1
        } else {
            return false
        }
    }
    return true
}
